### DISCUSSION TITLE
Interval not using Oracle specified Dialect

### DISCUSSION THREAD

**Discussion by indiVar0508** [sqlalchemy/sqlalchemy#10508] [Usage Questions]
Hi
I was working on a issue and was getting incorrect create table query for oracle, please refer below 

```py
# If i do this and create Table
table = Table('my_interval_table', metadata,
    Column('my_interval_column', sa.Interval(day_precision=9, second_precision=9))
)
# Query generated for Oracle DB
CREATE TABLE my_interval_table (my_interval_column INTERVAL DAY TO SECOND)

# But if i directly get INTERVAL from oracle dialect 
table = Table('my_interval_table', metadata,
    Column('my_interval_column', sa.dialects.oracle.types.INTERVAL(day_precision=9, second_precision=9))
)
# GIves query
CREATE TABLE my_interval_table (my_interval_column INTERVAL DAY(9) TO SECOND(9))
```

but for postgresDB it seems to work fine, in [Interval Doc](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Interval) it talks about postgres but not Oracle DB, so wanted to confirm.

Thanks.

**Comment by CaselIT** [comment]
Hi,

this seems like a bug to me. cc @zzzeek

**Reply by zzzeek** [reply]
sure, the equivalent arguments from the generic type should take effect

**Reply by CaselIT** [reply]
created https://github.com/sqlalchemy/sqlalchemy/issues/10509
